internal/local/k8s/cluster.go

```diff
@@ -0,0 +1,155 @@
+package k8s
+
+import (
+	"fmt"
+	"github.com/airbytehq/abctl/internal/local/localerr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+	"sigs.k8s.io/kind/pkg/cluster"
+	"time"
+)
+
+// Cluster is an interface representing all the actions taken at the cluster level.
+type Cluster interface {
+	// Create a cluster with the provided name.
+	Create(portHTTP int) error
```

Comment:
NIT: if we can find a way (that isn't annoying) to not require the port here, that would be cool, but def not a blocker

Reply:
Agreed.  Will revisit this in a follow-up PR.